server/src/main/java/org/apache/iotdb/db/metadata/MManager.java

```diff
@@ -548,13 +548,13 @@ public TSDataType getSeriesType(String path) throws MetadataException {
 
 
   /**
-   * Get all devices under given prefixPath.
+   * Get all device nodes under given prefixPath.
    *
    * @param prefixPath a prefix of a full path. if the wildcard is not at the tail, then each
    * wildcard can only match one level, otherwise it can match to the tail.
-   * @return A HashSet instance which stores devices names with given prefixPath.
+   * @return A list which stores device nodes with given prefixPath.
    */
-  public Set<String> getDevices(String prefixPath) throws MetadataException {
+  public List<MNode> getDevices(String prefixPath) throws MetadataException {
```

Comment:
Actually I think it's better to use `String` here so that you won't use `getFullPath()` when call this method ... What do you think?

Reply:
If using `String` here, we have to find the node again by path string in AlignByDeviceDataSet. So storaging MNode directly here can reduce that process.

Reply:
Sure. I didn't figure out a better way either... Maybe we could keep this then : )

Reply:
Actually It doesn't cost that much time finding device node. Like what you said, to keep the consistence with other methods, I deciced to change it to return List<String>.